nixos/modules/services/backup/burpServer.nix

```diff
@@ -0,0 +1,83 @@
+{ config, lib, networking, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.burpServer;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.burpServer = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable burp.";
+      };
+
+      user = mkOption {
+        default = "burp";
+        description = "The user to run burp as";
+      };
+
+      group = mkOption {
+        default = cfg.user;
+        description = "The group to run burp as (default same as user)";
+      };
+
+      configFile = mkOption {
+        default = "/etc/burp/burp.conf";
+        description = "Path to the burp config file. You need to provide a fully working configuration file and setup all paths yourself.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.burp;
+        example = literalExample "pkgs.burp";
+        description = ''burp package to use.'';
+      };
+
+    };
+
+  };
+
+  ###### implementation
+
+  config = mkIf config.services.burpServer.enable {
+
+    users.extraUsers.burp = {
```

Comment:
Thanks for the explanation regarding the user, that cleared up things for me.
You're right, it's probably "good enough" to just set the user and group both to "burp" and do not add configuration options for these.
Making that optional would take me too much time right now, as I'm still rather new to nix, so I'm just going to remove that option for now.